PyDSS/pyControllers/Controllers/MotorStall.py

```diff
@@ -5,94 +5,186 @@
+        :type FaultObj: class:`PyDSS.dssElement.dssElement`
+        :param Settings: A dictionary that defines the settings for the PvController.
+        :type Settings: dict
+        :param dssInstance: An :class:`opendssdirect` instance
+        :type dssInstance: :class:`opendssdirect`
+        :param ElmObjectList: Dictionary of all dssElement, dssBus and dssCircuit objects
+        :type ElmObjectList: dict
+        :param dssSolver: An instance of one of the classed defined in :mod:`PyDSS.SolveMode`.
+        :type dssSolver: :mod:`PyDSS.SolveMode`
+        :raises: AssertionError if 'RegulatorObj' is not a wrapped OpenDSS Regulator element
+
+    """
+        
     def __init__(self, MotorObj, Settings, dssInstance, ElmObjectList, dssSolver):
         super(MotorStall, self).__init__(MotorObj, Settings, dssInstance, ElmObjectList, dssSolver)
         self._class, self._name = MotorObj.GetInfo()
+        assert self._class == "Load", "Motor stall model can only be used with load models"
+        assert MotorObj.NumPhases == 1, "Motor stall model can only be used with single phase loads"
+        
         self.name = "Controller-{}-{}".format(self._class, self._name)
         self._ControlledElm = MotorObj
         self.__Settings = Settings
         self.__dssSolver = dssSolver
-
-        self._ControlledElm.SetParameter('model', 2)
+        self.__dssInstance = dssInstance
+        self.mode = 3
+        self.model_mode = self._ControlledElm.SetParameter('model', self.mode) # 3 - motor, 1 - Standard constant P+jQ
         self._ControlledElm.SetParameter('vminpu', 0.0)
 
         self.kw = self.__Settings['ratedKW']
         S = self.kw / self.__Settings['ratedPF']
         self.kvar = math.sqrt(S**2 - self.kw**2)
-        self._ControlledElm.SetParameter('kw', self.kw)
-        self._ControlledElm.SetParameter('kvar', self.kvar)
+        mode = 2
+        
+        if mode == 1:
+            self.kw_rated = self._ControlledElm.SetParameter('kw', self.kw)
+            self.kvar_rated = self._ControlledElm.SetParameter('kvar', self.kvar)        
+        elif mode == 2:
+            self.kw_rated = self._ControlledElm.GetParameter('kw')
+            self.kvar_rated = self._ControlledElm.GetParameter('kvar')
+        self.kva_rated = (self.kw_rated**2 + self.kvar_rated**2)**0.5
+            
         self.stall_time_start = 0
         self.stall = False
         self.disconnected =False
         self.Tdisconnect_start = 0
+        
+        self.R_stall_pu = self.__Settings['R_stall_pu']
+        self.kvbase = self._ControlledElm.sBus[0].GetVariable("kVBase")
+        self.Ibase = 1e3 * self.kva_rated /1e3 * self.kvbase
+
+        ######################################### TESTING CODE ###################################################
+        self.f = open(r"C:/Users/alatif/Desktop/NAERM/models/74012/Exports/results.csv", "w")
```

Comment:
Can this use the current working directory or the project directory?

Reply:
removed